components/selectors/parser.rs

```diff
@@ -17,6 +18,17 @@ use std::slice;
 use tree::SELECTOR_WHITESPACE;
 use visitor::SelectorVisitor;
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum SelectorParseError<T> {
+    PseudoElementInComplexSelector,
+    NoQualifiedNameInAttributeSelector,
+    TooManyCompoundSelectorComponents,
```

Comment:
Let's rename this to `TooManyCompoundSelectorComponentsInNegation`, or perhaps `TooComplexNegationSelector`?